tests/determinism.rs

```diff
@@ -0,0 +1,57 @@
+use anyhow::Result;
```

Comment:
Optional style nitpick: group imports into 3 blocks (std, external crates, this repo) like https://github.com/alan-turing-institute/uatk-spc/blob/main/src/init/commuting.rs. This is sadly not something `cargo fmt` or other tools enforce. I'm not sure it's that helpful for readers, and it certainly makes editing a little harder. Do you have a strong opinion here?

Reply:
I quite like this as a style choice and makes sense to be consistent across the repo. And reordering manually is straightforward enough given `cargo fmt` does not reorder after splitting into the three blocks. Updated here (8a3d559537b7dac832cdd6b0d606dde8be8b654f).